cfme/common/provider.py

```diff
@@ -410,31 +428,36 @@ def _all_available_policy_profiles(self):
     def _is_policy_profile_row_checked(self, row):
         return "Check" in row.find_element_by_xpath("../td[@width='16px']/img").get_attribute("src")
 
-    def num_template(self, db=True):
+    @variable(alias="db")
+    def num_template(self):
         """ Returns the providers number of templates, as shown on the Details page."""
-        if db:
-            ext_management_systems = cfmedb()["ext_management_systems"]
-            vms = cfmedb()["vms"]
-            truthy = True  # This is to prevent a lint error with ==True
-            temlist = list(cfmedb().session.query(vms.name)
-                           .join(ext_management_systems, vms.ems_id == ext_management_systems.id)
-                           .filter(ext_management_systems.name == self.name)
-                           .filter(vms.template == truthy))
-            return len(temlist)
-        else:
-            return int(self.get_detail("Relationships", self.template_name))
-
-    def num_vm(self, db=True):
+        ext_management_systems = cfmedb()["ext_management_systems"]
+        vms = cfmedb()["vms"]
+        truthy = True  # This is to prevent a lint error with ==True
+        temlist = list(cfmedb().session.query(vms.name)
+                       .join(ext_management_systems, vms.ems_id == ext_management_systems.id)
+                       .filter(ext_management_systems.name == self.name)
+                       .filter(vms.template == truthy))
```

Comment:
Actually, I think `# noqa` works here, I remember I have put it in DB queries already.